files/find.py

```diff
@@ -153,12 +155,19 @@
 '''
 
 def pfilter(f, patterns=None):
-    '''filter using glob patterns'''
     if patterns is None:
         return True
-    for p in patterns:
-        if fnmatch.fnmatch(f, p):
-             return True
+        
+    if type(patterns) == type([]):
```

Comment:
patterns are 'list' in the spec, no need to check here again as we always get a list